rcl_action/include/rcl_action/goal_state_machine.h

```diff
@@ -83,23 +80,38 @@ _set_canceled_event_handler(rcl_action_goal_state_t state, rcl_action_goal_event
 }
 
 // Transition map
-rcl_action_goal_event_handler
-  _goal_state_transition_map[GOAL_STATE_NUM_STATES][GOAL_EVENT_NUM_EVENTS] = {
-  [GOAL_STATE_ACCEPTED] = {
-    [GOAL_EVENT_EXECUTE] = _execute_event_handler,
-    [GOAL_EVENT_CANCEL] = _cancel_event_handler,
-  },
-  [GOAL_STATE_EXECUTING] = {
-    [GOAL_EVENT_CANCEL] = _cancel_event_handler,
-    [GOAL_EVENT_SET_SUCCEEDED] = _set_succeeded_event_handler,
-    [GOAL_EVENT_SET_ABORTED] = _set_aborted_event_handler,
-  },
-  [GOAL_STATE_CANCELING] = {
-    [GOAL_EVENT_SET_SUCCEEDED] = _set_succeeded_event_handler,
-    [GOAL_EVENT_SET_ABORTED] = _set_aborted_event_handler,
-    [GOAL_EVENT_SET_CANCELED] = _set_canceled_event_handler,
-  },
-};
+static rcl_action_goal_event_handler
+  _goal_state_transition_map[GOAL_STATE_NUM_STATES][GOAL_EVENT_NUM_EVENTS];
+
+static inline void
+rcl_action_goal_transition_map_init()
+{
+  // Only run this function once
+  static bool _is_goal_state_transition_map_init = false;
+  if (_is_goal_state_transition_map_init) {
```

Comment:
If this could be called from multiple threads, there is a TOCTTOU bug here; thread 1 could have read the state of `_is_goal_state_transition_map_init` as false then been swapped out, then thread 2 could read the state of `_is_goal_state_transition_map` as false, and do the initialization.  Then when thread 1 wakes back up, it still thinks that it needs to do the initialization, and does it again.

Reply:
Moving the initialization to the C file let me do a direct initialization, so I believe this issue goes away.